### ISSUE TITLE
Impossible to redirect to the good resource

### ISSUE THREAD

**Issue by leobury** [CharrafiMed/global-search-modal#70]
Hey,

Suddenly it's impossible for all my users to save both history and favourites. Do you have any ideas? 
I've tried updating and it doesn't work.

When I click on the saved item, I'm redirected right to the resource. This is what I have in my local storage:
<img width="559" alt="image" src="https://github.com/user-attachments/assets/9ac6607c-1943-43c6-9750-5eb9013da578">

**Comment by CharrafiMed** [comment]
> Hey,
> 
> Suddenly it's impossible for all my users to save both history and favourites. Do you have any ideas? I've tried updating and it doesn't work.
> 
> When I click on the saved item, I'm redirected right to the resource. This is what I have in my local storage: <img alt="image" width="559" src="https://private-user-images.githubusercontent.com/102532207/376063422-9ac6607c-1943-43c6-9750-5eb9013da578.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4OTI0MDksIm5iZiI6MTcyODg5MjEwOSwicGF0aCI6Ii8xMDI1MzIyMDcvMzc2MDYzNDIyLTlhYzY2MDdjLTE5NDMtNDNjNi05NzUwLTVlYjkwMTNkYTU3OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNFQwNzQ4MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDZmOTI1NzI3MzA4YzY1YTA4NDJmNDBhYTJiNDMwNTljYmI0ZmFmNDU5NGZhOGMwNWFjYWIwMmMwMTI0ZTM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.v4BX3pZ7LJz4ib0jflA4aVovdzHsIC_85q6sthCWR2E">

hey! I didn't see the URL key in your local storage?

**Comment by leobury** [comment]
> > Hey,
> > Suddenly it's impossible for all my users to save both history and favourites. Do you have any ideas? I've tried updating and it doesn't work.
> > When I click on the saved item, I'm redirected right to the resource. This is what I have in my local storage: <img alt="image" width="559" src="https://private-user-images.githubusercontent.com/102532207/376063422-9ac6607c-1943-43c6-9750-5eb9013da578.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4OTI0MDksIm5iZiI6MTcyODg5MjEwOSwicGF0aCI6Ii8xMDI1MzIyMDcvMzc2MDYzNDIyLTlhYzY2MDdjLTE5NDMtNDNjNi05NzUwLTVlYjkwMTNkYTU3OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNFQwNzQ4MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDZmOTI1NzI3MzA4YzY1YTA4NDJmNDBhYTJiNDMwNTljYmI0ZmFmNDU5NGZhOGMwNWFjYWIwMmMwMTI0ZTM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.v4BX3pZ7LJz4ib0jflA4aVovdzHsIC_85q6sthCWR2E">
> 
> hey! I didn't see the URL key in your local storage?

Yes its the problem.. When we add something in favorites, the url is not saved

**Comment by CharrafiMed** [comment]
Did you republish your assets and clear the local storage? I think this issue is not a bug, just something in your panels! if  not we can discuss more about this issue

**Comment by leobury** [comment]
I've just run a few tests. The problem occurs when you set a recent search as favourite.
Recent searches contain a url, but when I change it to favourite, the url is lost :/

**Comment by CharrafiMed** [comment]
emmm oki  i will play with it

**Comment by jkeppens-zenchef** [comment]
@CharrafiMed any updates on this?

**Comment by CharrafiMed** [comment]
Hi @jkeppens-zenchef 

I haven't had the time to revisit this issue recently. The last time I tried reproducing the bug, it didn’t occur on my setup. Given that, I’d highly encourage you to explore the issue. If you can pinpoint the cause or even create a fix, a pull request would be greatly appreciated!

thank you.

**Comment by jkeppens-zenchef** [comment]
I think I found it:

`resources/views/vendor/global-search-modal/components/search/summary/summary-wrapper.blade.php:`

`clickFunction="addToFavorites(result.item, result.group)"`

result.url should be passed as 3rd param:

`addToFavorites(result.item, result.group, result.url)`

**Comment by jkeppens-zenchef** [comment]
@CharrafiMed looks like I do not have the rights to push a branch + make PR to this repo, so to fix this, the change needed is:

`global-search-modal/resources/views/components/search/summary/summary-wrapper.blade.php`

line 28:

replace `clickFunction="addToFavorites(result.item, result.group)"` by: `clickFunction="addToFavorites(result.item, result.group, result.url)"`

**Comment by jkeppens-zenchef** [comment]
PR: [https://github.com/CharrafiMed/global-search-modal/pull/88/files](https://github.com/CharrafiMed/global-search-modal/pull/88/files)

**Comment by jkeppens-zenchef** [comment]
@CharrafiMed PR: [https://github.com/CharrafiMed/global-search-modal/pull/88/files](https://github.com/CharrafiMed/global-search-modal/pull/88/files)

**Comment by CharrafiMed** [comment]
@jkeppens-zenchef  thanks, merged
